### ISSUE TITLE
Auto establish context/pathing better

### ISSUE THREAD

**Issue by mulligandev** [rooseveltframework/roosevelt#553]
Currently, when nesting roosevelt as a sub app pathing is a little funky and requires some extra tweaks to get it to work. We should be able to isolate roosevelt a little bit better for those cases to make it easier to get up and running with sub apps.

Normally when using our views we could simply use:
```
<link rel='stylesheet' href='/css/styles.css'>
``` 
to include a file within our html. But when using [sub apps it requires for the path to be relative ](https://github.com/mulligandev/roosevelt-test-apps/blob/237e006fdb3d1cbec95b5493c40614dd701ac545/sub-app/subApp/mvc/views/layouts/main.html#L9)(unlike the main app) so any links to files need to have a `./` prefix not just `/`.
For example:
```
<link rel='stylesheet' href='./css/styles.css'>
```

Also, sub apps required us to [explicitly](https://github.com/mulligandev/roosevelt-test-apps/blob/237e006fdb3d1cbec95b5493c40614dd701ac545/sub-app/subApp/app.js#L8) specify the `appDir` otherwise it will be sourced from the parent require. 

Finally, we should consider the html validator and auto killer when using sub apps. Currently for the sub apps we have to [set the autoKiller to false](https://github.com/mulligandev/roosevelt-test-apps/blob/237e006fdb3d1cbec95b5493c40614dd701ac545/sub-app/subApp/roosevelt.json#L21) otherwise it throws an error. ~~Also, unless there is a detached validator already running, the sub app will not work.~~
~~Example:~~
  - 1.) ~~Make sure there is no html validator running~~
  - 2.) ~~`npm run dev`~~
  - 3.) ~~`http://localhost:43711/foo/` will throw a 404~~
  - 4.) ~~Kill that app and spin it up again (the validator will still be running)~~
  - 5.) ~~`http://localhost:43711/foo/` will render properly~~

[Here is a repo that demonstrates a main app and sub app](https://github.com/mulligandev/roosevelt-test-apps/tree/master/sub-app)

In the next version of roosevelt we could have an optional parameter that such as `"childApp": true` or something to that effect that helps isolate the sub app and disables unwanted features that may already be enabled in the main application. Another idea is that if the main app and sub app both share the same directory structures (mvc, statics, app.js, etc..) we could make the sub app use the same roosevelt config as the main app but still keep it isolated properly.

**Updated 8/5/19**
- Broken links have been updated
- Text with strikethrough can no longer be reproduced

**Comment by kethinov** [comment]
After offline discussion we concluded there isn't an obvious way to do this and largely comes with the territory of using the routers feature.

The tl;dr of the problem is this: when using sub apps / routers, it constrains the public folder of the sub app as follows:

1. `/paths` inherit the path of the parent, changing `http://domain/path` to `http://domain/parentApp/path`, which will likely break the sub app's static assets.
2. `./paths` will generally still work because they are relative, but this is often not desired in production contexts.
3. `http://domain/absolutePaths` will still work.

The hope in filing this issue was to try to find a way to keep 1. still working, but we don't think that is possible.

Doing 3. in production contexts if the app is meant to be consumed as a sub app is probably the best you can do.

As such, the best we can do is document this limitation in the README and call it good.
